### PR TITLE
Tile Inspector UX Improvements

### PR DESCRIPTION
**Pull request by Broxzier**
This PR makes the list in the Tile Inspector more consistent with other windows in terms of looks. Black text, no dark/light for odd/even rows, white text on hover, and darker background colour.

The UX has also been improved, by making the window automatically resize when clicking on something from the list, as suggested by #5832, and by allowing the player to Ctrl+Click on tile elements to automatically select them.

This PR is also aimed to make the code a bit more flexible. It removes two of the intents that were made for it, automatically sets the correct page, and the pressed widgets get reset on switching. With these changes, two small issues have been fixed (after pasting an element, the page for the pasted element would not be shown, and seemingly at random buttons stay pressed).

### CODE REVIEW

File: distribution/changelog.txt [hunk 1]
```diff
@@ -33,10 +33,12 @@
 - Fix: [#7544] Starting a headless server with no arguments causes the game to freeze.
 - Improved: [#2989] Multiplayer window now changes title when tab changes.
 - Improved: [#5339] Change eyedropper icon to actual eyedropper and change cursor to crosshair.
+- Improved: [#5832] Resize tile inspector automatically when selecting a tile element.
 - Improved: [#7302] Raising land near the map edge makes the affected area smaller instead of showing an 'off edge map' error.
 - Improved: [#7435] Object indexing now supports multi-threading.
 - Improved: [#7510] Add horizontal clipping to cut-away view options.
 - Improved: [#7531] "Save track design" dropdown now stays open.
+- Improved: [#7548] Add Ctrl + Click functionality to select tile elements with the tile inspector.
```
Comment (AaronVanGeffen) [inline distribution/changelog.txt]: Could you clarify here how this new function differs from the old one?

**Comment by AaronVanGeffen** [comment]
As this PR changes the intent list, I believe it should increment the network version as well.

File: distribution/changelog.txt (hunk 1 above)
Comment (Broxzier) [reply distribution/changelog.txt]: I find it difficult to summerize it in few words. The current way of selecting elements is first clicking on the tile, then on the element in the list. With this you can Ctrl+click on the element in the viewport directly, and it will select the tile that the element is on and select the element in the list.

Comment (AaronVanGeffen) [reply distribution/changelog.txt]: Ah, that helps. Perhaps something like "Ctrl-clicking an element now directly selects it and its tile in the tile inspector."? Although that suggests you don't need to have the tile inspector open, perhaps.

On a side note, I think we've substituted Ctrl shortcuts with Cmd ones on macOS. Is that taken into account here?

Comment (Broxzier) [reply distribution/changelog.txt]: Thanks, that's better already. "Ctrl-Clicking with the tile inspector open now now directly selects the element and its tile." Perhaps using "Ctrl/Cmd + Clicking" instead it clearer for MacOS users? Or is it normal to refer to the Cmd key with Ctrl?

The code cheks for `PLACE_OBJECT_MODIFIER_COPY_Z` rather than for Ctrl input directly.

Comment (AaronVanGeffen) [reply distribution/changelog.txt]: > The code cheks for `PLACE_OBJECT_MODIFIER_COPY_Z` rather than for Ctrl input directly.

Ah, I don't think that's been adjusted for macOS. Should be fine to just refer to Ctrl, then. So, "Ctrl-clicking with the tile inspector open now directly selects an element and its tile."?

Comment (Broxzier) [reply distribution/changelog.txt]: Thanks! I've amended the commit.

**Comment by Broxzier** [comment]
@Gymnasiast I don't see how this is different from the load/save window, and zebra stripes were removed from that list.

**Comment by AaronVanGeffen** [comment]
For what it's worth, I also think the zebra stripes aren't necessarily for this window.

**Comment by Gymnasiast** [comment]
> @Gymnasiast I don't see how this is different from the load/save window, and zebra stripes were removed from that list.

And? Doesn't prove that it improves things, just that more dialogs have been ****ed up in order to be "more consistent". But even it was a good thing for the load/save window (and I doubt that!), the tile inspector has more columns, making the benefits of zebra striping bigger.

**Comment by Broxzier** [comment]
The way you say it is as if you're attacking me for daring to make that change. My argument was meant to explain why I made the change. Since the zebra stripes were removed from the load/save window, the tile inspector and title sequence editor are the only ones still using them, so I figured the majority would be in favour of them being removed. Personally I'm in favour of keeping them. In fact I'd like to see them used for all other lists (rides, guests, staff, load/save, title sequence editor, and probably some others). If the majority agrees, I'm more than happy to reintroduce them.

**Comment by Broxzier** [comment]
I made it exactly like the window from #7420 (right), and then a version with less contrast between the rows (left). I went with the left one.

image

@Gymnasiast ping.

**Comment by Gymnasiast** [comment]
I prefer the right one. Perhaps a contrast value that is the average between the two would be an idea?

**Comment by AaronVanGeffen** [comment]
Hmm, I personally prefer the left one. The left one already uses some dithering techniques to get its  colours. I'm afraid combining the left and right versions will lead to the window [looking like this](https://user-images.githubusercontent.com/604665/38827938-c3efe6b2-41b4-11e8-9387-11ec1f49861e.png), which makes the text look too fuzzy to me.

**Comment by Gymnasiast** [comment]
Right, I'll comprise. Let's go with the left one then.

File: src/openrct2-ui/WindowManager.cpp [hunk 2]
```diff
@@ -343,20 +343,6 @@ class WindowManager final : public IWindowManager
             window_tile_inspector_clear_clipboard();
             break;
 
-        case INTENT_ACTION_SET_TILE_INSPECTOR_PAGE:
```
Comment (Gymnasiast) [inline src/openrct2-ui/WindowManager.cpp]: Why was this removed?

Comment (Broxzier) [reply src/openrct2-ui/WindowManager.cpp]: > It removes two of the intents that were made for it, **automatically sets the correct page**, and the pressed widgets get reset on switching.
